### ISSUE TITLE
About the multi-class problem

### ISSUE THREAD

**Issue by Inosonnia** [milesial/Pytorch-UNet#174]
Thanks for your code.
I implemented this code to solve a simple multi-class segmentation problem, and encounter several problems.

1. For image / mask pair, I prepare a RGB-image (3 * H * W) set and a corresponding mask set (1 * H * W).
2. As I have 4 classes, the label of each mask pixel is selected from [1, 2, 3, 4].
3. I set n_channels=3 and n_classes=3.

When I start to train:
1. SpatialClassNLLCriterion.cu:104: void cunn_SpatialClassNLLCriterion_updateOutput_kernel(T *, T *, T *, long *, T *, int, int, int, int, int, long) [with T = float, AccumT = float]: block: [0,0,0], thread: [544,0,0] Assertion `t >= 0 && t < n_classes` failed. 
----- It seems that the numeric value of mask should not be greater than 1, thus integer labels within [1, 2, 3, 4] are not feasible.
As the consequence, I have the mask divide img_trans.max() as the image do, thus the label is [0.25, 0.5, 0.75, 1].
This action makes the training procedure fine (Validation cross entropy: 1.005), but the 100-epoch module predicts a all [0.25] mask image.

I read the related issue (https://github.com/milesial/Pytorch-UNet/issues/15), but get no useful info. (It seems that all related issues are solved by the most recent commit).

Looking forward for your reply, thanks again.

**Comment by milesial** [comment]
Hi,
If you have 4 classes, why not set n_classes=4 ?

The labels should be [0, 1, 2, 3], as described in the [documentation](https://pytorch.org/docs/stable/nn.html?highlight=crossentropyloss#torch.nn.CrossEntropyLoss) for the CE loss.

![image](https://user-images.githubusercontent.com/30204471/81460756-3aef4700-915c-11ea-8348-d80c01efcd16.png)

**Comment by milesial** [comment]
Go to https://github.com/milesial/Pytorch-UNet/issues/15 for multi class discussion
